src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -126,6 +129,30 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		this._register(this._onDeleteConnectionProfile);
 	}
 
+	/**
+	 * Set the initial value for the connection provider map and listen to the provider change event
+	 */
+	private initializeConnectionProvidersMap() {
+		this.setConnectionProvidersMap();
+		if (this._capabilitiesService) {
+			this._capabilitiesService.onCapabilitiesRegistered(() => {
```

Comment:
Can't this be simplified to
```typescript
this._capabilitiesService.onCapabilitiesRegistered(this.setConnectionProvidersMap)
```
The signature of lambda is the same as the method it calls...

Reply:
No it cannot, it expects a argument type ProviderFeatures and by just doing this it complains that we cannot assign void.

Reply:
Signature of lambda is not void

Reply:
Oh so it allows you to discard the arguments when defining lambda (because signature is clearly ```void ()``` for the anonymous delegate here) and doesn't do that when you pass in a function reference? Lame. :\